app/Services/LengthValidatorService.php

```diff
@@ -0,0 +1,40 @@
+<?php
+
+namespace App\Services;
+
+use App\Models\Calendar;
+use App\Http\Requests\ValidateLength as Request;
+
+class LengthValidatorService
+{
+    /**
+     * @return array
+     */
+    public function validateLength(Calendar $calendar, Request $request)
+    {
+        $day = $request['day'] ?? 0;
+        $month = $request['month'];
+        $length = $request['length'];
+
+        $daysInYear = $calendar->daysInYear();
+        $counter = 0;
+        $monthLength = 0;
+        foreach ($calendar->monthDataProperties() as $monthData) {
+            $counter = $counter + 1;
+            if ($counter >= $month) {
+                $monthLength = $monthLength + $monthData['data-length'];
+            }
+        }
+        $totalLength = $monthLength - $day + $daysInYear;
+        if ($length >= $totalLength) {
+            return json_encode([
```

Comment:
Shouldn't be the service's job to return an encoded json, but the controller :-)